### PR TITLE
Add grpc_pollset_set to grpc_resolve_address

### PR DESCRIPTION
**Pull request by y-zeng**
As grpc_resolve_address is an async interface, the caller should provide a grpc_pollset_set.

Part of #7771 

---
Tests are failing with known issues:
Basic Tests Linux: #9007

### CODE REVIEW

File: test/core/iomgr/resolve_address_test.c [hunk 1]
```diff
@@ -46,16 +48,72 @@ static gpr_timespec test_deadline(void) {
...
-void args_finish(args_struct *args) {
+void args_finish(grpc_exec_ctx *exec_ctx, args_struct *args) {
   GPR_ASSERT(gpr_event_wait(&args->ev, test_deadline()));
   grpc_resolved_addresses_destroy(args->addrs);
+  grpc_pollset_set_del_pollset(exec_ctx, args->pollset_set, args->pollset);
+  grpc_pollset_set_destroy(args->pollset_set);
+  grpc_closure do_nothing_cb;
+  grpc_closure_init(&do_nothing_cb, do_nothing, NULL);
+  grpc_pollset_shutdown(exec_ctx, args->pollset, &do_nothing_cb);
+  // exec_ctx needs to be flushed before calling grpc_pollset_shutdown
```
Comment (y-zeng) [inline test/core/iomgr/resolve_address_test.c]: Yes, it should be `grpc_pollset_destroy()`.
Fixed. Thanks a lot for the review!

File: src/core/ext/client_channel/resolver.h [hunk 2]
```diff
@@ -36,13 +36,15 @@
...
 #include "src/core/lib/iomgr/iomgr.h"
+#include "src/core/lib/iomgr/pollset_set.h"
 
 typedef struct grpc_resolver grpc_resolver;
 typedef struct grpc_resolver_vtable grpc_resolver_vtable;
 
 /** \a grpc_resolver provides \a grpc_channel_args objects to its caller */
 struct grpc_resolver {
   const grpc_resolver_vtable *vtable;
+  grpc_pollset_set *pollset_set;
```
Comment (y-zeng) [inline src/core/ext/client_channel/resolver.h]: #8795 looks awesome! That makes it possible to add `grpc_pollset_set` as a parameter to `grpc_resolver_create()`, as the resolver will only be instantiated inside client_channel.c.
Will wait for #8795 to be merged.

Comment (y-zeng) [inline src/core/ext/client_channel/resolver.h]: Removed `pollset_set` from the `grpc_resolver` struct, added it as a parameter of `grpc_resolver_create()`.

**Comment by y-zeng** [comment]
Updated with #8795

**Comment by dgquintas** [comment]
@y-zeng could you update the documentation for the addition of the new pollset argument in src/core/ext/client_channel/resolver_registry.h?

**Pull request merged**